tests/build/source_build.go

```diff
@@ -67,21 +73,118 @@ func CheckParentSources(c client.Client, tektonController *tekton.TektonControll
 
 	allowed, err := build.IsImagePulledFromAllowedRegistry(imageWithoutTag)
 	Expect(err).ShouldNot(HaveOccurred())
-	if !allowed {
-		Expect(buildResult.BaseImageSourceIncluded).Should(BeFalse())
-		containsLog, err := build.SourceBuildTaskRunLogsContain(
-			tektonController, pr,
-			fmt.Sprintf("Image %s does not come from supported allowed registry", lastBaseImage),
-		)
-		Expect(err).ShouldNot(HaveOccurred())
-		Expect(containsLog).Should(BeTrue())
-		return
-	}
 
-	parentSourceImage, err := build.ResolveSourceImage(imageWithoutTag)
+	var parentSourceImage string
+
+	if allowed {
+		parentSourceImage, err = build.ResolveSourceImageByVersionRelease(imageWithoutTag)
+	} else {
+		parentSourceImage, err = build.ResolveKonfluxSourceImage(imageWithoutTag)
+	}
 	Expect(err).ShouldNot(HaveOccurred())
+
 	allIncluded, err := build.AllParentSourcesIncluded(parentSourceImage, buildResult.ImageUrl)
-	Expect(err).ShouldNot(HaveOccurred())
+
+	if err != nil {
+		msg := err.Error()
+		if strings.Contains(msg, "parent source image manifest") && strings.Contains(msg, "MANIFEST_UNKNOWN:") {
+			return
+		} else {
+			Fail(fmt.Sprintf("failed to check parent sources: %v", err))
+		}
+	}
+
 	Expect(allIncluded).Should(BeTrue())
 	Expect(buildResult.BaseImageSourceIncluded).Should(BeTrue())
 }
+
+func CheckSourceImage(srcImage, gitUrl string, hub *framework.ControllerHub, pr *pipeline.PipelineRun) {
+	//Check if hermetic enabled
+	isHermeticBuildEnabled := build.IsHermeticBuildEnabled(pr)
+	GinkgoWriter.Printf("HERMETIC STATUS: %v\n", isHermeticBuildEnabled)
+
+	//Get prefetch input value
+	prefetchInputValue := build.GetPrefetchValue(pr)
+	GinkgoWriter.Printf("PRE-FETCH VALUE: %v\n", prefetchInputValue)
+
+	filesExists, err := build.IsSourceFilesExistsInSourceImage(
+		srcImage, gitUrl, isHermeticBuildEnabled, prefetchInputValue)
+	Expect(err).ShouldNot(HaveOccurred())
+	Expect(filesExists).To(BeTrue())
+
+	c := hub.CommonController.KubeRest()
+	CheckParentSources(c, hub.TektonController, pr)
+}
+
+func updateParentImageInChildTestRepo(parentImageUrl, childTestRepoUrl string) {
+	childRepoUrl, err := url.Parse(childTestRepoUrl)
+	Expect(err).Should(Succeed())
+
+	org, repo := path.Split(childRepoUrl.Path)
+	gh, err := build.NewGithubClient(strings.Trim(org, "/"))
+	Expect(err).Should(Succeed())
+	dockerfile, err := gh.GetFile(repo, "Dockerfile", "main")
+	Expect(err).Should(Succeed())
+
+	dockerfileContent, err := dockerfile.GetContent()
+	Expect(err).Should(Succeed())
+	Expect(dockerfileContent).ShouldNot(BeEmpty())
+
+	newFrom := fmt.Sprintf("FROM %s", parentImageUrl)
+	updatedContent := strings.ReplaceAll(dockerfileContent, "FROM scratch", newFrom)
```

Comment:
What test repo will this be used for? I still don't really understand why we need to replace the `FROM` dynamically.

Why can't it be `FROM <some-image-that-we-build-once-and-never-change>`?

Reply:
The image built from a test repo by a build-definitions CI job is bound to the build-definitions' last commit hash, it can't be pinned. I also would like to set the image built from a test repo under redhat-appstudio-qe rather than others e.g. my personal github org.

But now, I'm considering the way you mentioned either. Asking for help from QE to see which Konflux (prod/staging/qe) I can onboard the test repo to in order to build the image `<some-image-that-we-build-once-and-never-change>`.